src/helpers/current-page-helper.php

```diff
@@ -228,9 +229,10 @@ public function is_home_static_page() {
 	public function is_static_posts_page() {
 		$wp_query = $this->wp_query_wrapper->get_main_query();
 
-		$page_for_posts = (int) \get_option( 'page_for_posts' );
-
-		return ( $page_for_posts > 0 && $page_for_posts === $wp_query->get_queried_object_id() );
+		return (
+			$wp_query->is_posts_page &&
+			\is_a( $wp_query->get_queried_object(), WP_Post::class )
```

Comment:
This can literally only be true when the post type is `page`, whereas this allows for more cases where this is true?

Reply:
Updated to also do an explicit check for the post_type of the queried object. I'm not sure if the situation can ever occur where `is_posts_page` is ever true and the queried object is a post but not a page but given possible interactions with other plugins it certainly doesn't hurt to check.